### ISSUE TITLE
Document and verify pymanopt's Positive manifold composes with GMM(penalty=...)

### ISSUE THREAD

**Issue by ligon** [ligon/ManifoldGMM#31]
## Motivation

Came out of a design conversation following #19 MR1 (parameter-penalty hook) and the K-Aggregators exp-link runaway (per the diag log on #19's empirical comment).  Empirically, the runaway is **weak identification** of the c-index, not a positivity-constraint problem -- the exp link already enforces positivity.  But thinking through how a reader would interpret the API surface, two questions surface:

1. **The exp-link parameterisation is, formally, the log chart on the positive-reals manifold.** When K-Aggregators uses `theta_j = exp(c_0 + sum c_m zeta_{jm})` and penalises `||c[1:]||^2` (Tikhonov on the index slopes), that *is* the Riemannian penalty on `log theta_j` -- ℝ⁺ with the affine-invariant metric is flat under log, so the Euclidean penalty on chart coordinates is the manifold-Riemannian penalty.  But this equivalence is currently nowhere in the documentation.  A reader looking at `GMM(restriction, penalty=...)` doesn't see the manifold-native framing.

2. **For consumers wanting positivity-constrained parameters as primary (not as the image of an exp link), pymanopt has the right manifold built-in.**  `pymanopt.manifolds.positive.Positive(p, 1)` implements element-wise strict positivity with the affine-invariant Riemannian metric `<u, v>_theta = sum u_j v_j / theta_j^2` (Fisher information for scale).  The retraction `theta -> theta * exp(v / theta)` is strictly positive by construction.  `Product([Euclidean(k), Positive(p, 1)])` is the natural composition for mixed unconstrained + positive parameters.

ManifoldGMM's `Manifold.from_pymanopt(...)` wraps arbitrary pymanopt manifolds, so library-side this composes today **with zero changes**.  But there's no docstring example showing it, and no integration test verifying it end-to-end with `GMM(penalty=...)`.

## What this issue tracks (library-side, in scope)

Three small deliverables, all confined to ManifoldGMM:

1. **Verify the import path** `from pymanopt.manifolds.positive import Positive` resolves cleanly against the pymanopt pinned in `pyproject.toml`.  Two-minute check.
2. **Integration test**: `tests/econometrics/test_positive_manifold.py` (new file).  Fit a synthetic problem with a positive-parameter `Positive(p, 1)` manifold (or `Product([Euclidean(k), Positive(p, 1)])` for the mixed case), optional `penalty=...` (Riemannian-style penalty on `log theta`), and verify:
   - Convergence to a known synthetic optimum.
   - All replicate `theta_j` are strictly positive throughout the trajectory (retraction never crosses the boundary).
   - `penalty=None` and `penalty=callable` both compose cleanly with the `Positive` manifold.
   - `MomentWildBootstrap` replicates on a `Positive` fit (#29's penalty-propagation fix in scope) all stay positive too.
3. **Docstring example** on `PenaltyStrategy` (or `Manifold.from_pymanopt`) showing the canonical pattern:
   ```python
   manifold = Manifold.from_pymanopt(
       Product([Euclidean(k), Positive(p, 1)])
   )
   gmm = GMM(
       restriction_with_positive_block(manifold),
       penalty=lambda theta: lam * jnp.sum(jnp.log(theta[1])**2),
   )
   ```
   Plus a one-line note that the Euclidean-on-`log theta` penalty IS the Riemannian penalty on the `Positive` manifold under the affine-invariant metric.

Total work: probably ~100-150 LOC including the test file.

## What's out of scope

**Per-product theta_j on (R^+)^N as primary parameters** -- the substantive estimator reformulation discussed in the original thread.  This would change which moments identify which parameters: under the exp-link approach, all c-coefficients are identified by all moments jointly (cross-product extrapolation via the index); under per-product theta_j on `Positive`, each theta_j is identified by its own moments only.  Trade-off:

- **Pro**: no cross-product runaway -- weakly-identified products give wide CIs but don't drag others.
- **Con**: more parameters (N_products instead of 6 c-coefficients in the K-Aggregators case); typically wider CIs per product without the shared-slope pooling.
- **Where**: this is a K-Aggregators-side estimator redesign (rewriting `gi_jax_exp` in `cereal_gmm_estimate.py`), not a ManifoldGMM library change.  File a K-Aggregators-side issue if/when there's appetite to try it.

ManifoldGMM library-side enables this redesign with no further changes -- the deliverables above suffice as the integration story.

## Penalty form question (briefly)

A reader of the canonical example above might ask: why penalise `log theta_j` in Euclidean form rather than via a Riemannian-distance helper?  Answer: under the `Positive` manifold's affine-invariant metric, the geodesic distance is `d(theta, theta_ref) = |log theta - log theta_ref|`.  Penalising `(log theta_j)^2` IS the squared geodesic distance from 1.  This is the manifold-Riemannian penalty by another name.  The `PenaltyStrategy` Protocol's optional `hessian_tangent(theta, basis)` method (#19 MR1) is the right surface for callers who want to supply an analytic Hessian in the manifold's tangent basis -- a docstring note can point this out without further code.

## Cross-references

- **#19** (parent: parameter-penalty hook; this issue is about a manifold + penalty composition story that #19's surface already supports).
- **#29** (closed, PR #30): `MomentWildBootstrap` propagates the penalty into replicates -- the integration test should exercise the `Positive` + clustered + penalised bootstrap as a regression test for #29's fix on a non-Euclidean parameter manifold.
- Pymanopt source: `pymanopt/manifolds/positive.py` for the `Positive` manifold's metric and retraction.

## Severity

Low.  This is a documentation + integration-test capture of an already-working composition, motivated by the fact that K-Aggregators (the only known consumer with a positivity-constrained parameter) currently uses the exp-link parameterisation and would have to re-design its moment specification to switch to the `Positive` manifold.  No active blocker.
